whatsapp/client.go

```
@@ -291,60 +291,98 @@ func (c *Client) SendTextMessage(ctx context.Context, chatJID string, text strin
+// local path: absolute, "~/"-relative, "file://" or relative to the process
+// working directory. A relative path that does not exist is rejected outright
+// instead of being handed to the HTTP client as if it were a hostname.
 func (c *Client) readMediaSource(source string) ([]byte, string, error) {
 	const maxSize = 16 * 1024 * 1024
 
-	// Expand ~ to home directory
-	if strings.HasPrefix(source, "~/") {
-		home, err := os.UserHomeDir()
+	if strings.HasPrefix(source, "http://") || strings.HasPrefix(source, "https://") {
+		c.log.Infof("Downloading from URL: %s", source)
+		resp, err := mediaHTTPClient.Get(source)
```

golangci-lint (2.12.2)

[error] (*net/http.Client).Get must not be called. use (*net/http.Client).Do(*http.Request)

(noctx)